src/devices/src/virtio/vsock/unix/muxer.rs

```diff
@@ -1379,4 +1416,73 @@ mod tests {
         // not be any pending RX in the muxer.
         assert!(!ctx.muxer.has_pending_rx());
     }
+
+    #[test]
+    fn test_vsock_basic_metrics() {
+        // Save the metrics values that we need tested.
+        let mut tx_packets_count = METRICS.vsock.tx_packets_count.count();
+        let mut rx_packets_count = METRICS.vsock.rx_packets_count.count();
+
+        let tx_bytes_count = METRICS.vsock.tx_bytes_count.count();
+        let rx_bytes_count = METRICS.vsock.rx_bytes_count.count();
+
+        let conns_added = METRICS.vsock.conns_added.count();
+        let conns_removed = METRICS.vsock.conns_removed.count();
+
+        // Create a basic connection.
+        let mut ctx = MuxerTestContext::new("vsock_basic_metrics");
+        let peer_port = 1025;
+        let (mut stream, local_port) = ctx.local_connect(peer_port);
+
+        // Once the handshake is done, we check that the TX bytes count has
+        // not been increased.
+        assert_eq!(METRICS.vsock.tx_bytes_count.count(), tx_bytes_count);
+
+        // Check that one packet was sent through the handshake.
+        assert_eq!(METRICS.vsock.tx_packets_count.count(), tx_packets_count + 1);
+        tx_packets_count = METRICS.vsock.tx_packets_count.count();
+
+        // Check that one packet was received through the handshake.
+        assert_eq!(METRICS.vsock.rx_packets_count.count(), rx_packets_count + 1);
+        rx_packets_count = METRICS.vsock.rx_packets_count.count();
+
+        // Check that a new connection was added.
+        assert_eq!(METRICS.vsock.conns_added.count(), conns_added + 1);
+
+        // Send some data from guest to host.
+        let data = [1, 2, 3, 4];
+        ctx.init_data_pkt(local_port, peer_port, &data);
+        ctx.send();
+
+        // Check that tx_bytes was incremented.
+        assert_eq!(
+            METRICS.vsock.tx_bytes_count.count(),
+            tx_bytes_count + data.len()
+        );
+
+        // Check that one packet was accounted for.
+        assert_eq!(METRICS.vsock.tx_packets_count.count(), tx_packets_count + 1);
+
+        // Send some data from the host to the guest.
+        let data = [5, 6, 7, 8];
```

Comment:
nit: can we make here a slice of a len different than 4? maybe `METRICS.vsock.rx_bytes_count.count()` is equal with `rx_bytes_count + data.len()` because of the data that was sent before from guest to host, not vice versa.

Reply:
Increased the buffer size.